modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -4278,20 +4278,36 @@ void GDScriptAnalyzer::reduce_preload(GDScriptParser::PreloadNode *p_preload) {
 			}
 		} else {
 			// TODO: Don't load if validating: use completion cache.
+			String resource_type = ResourceLoader::get_resource_type(p_preload->resolved_path);
 
 			// Must load GDScript separately to permit cyclic references
 			// as ResourceLoader::load() detects and rejects those.
-			if (ResourceLoader::get_resource_type(p_preload->resolved_path) == "GDScript") {
+			if (resource_type == "GDScript") {
 				Error err = OK;
 				Ref<GDScript> res = GDScriptCache::get_shallow_script(p_preload->resolved_path, err, parser->script_path);
 				p_preload->resource = res;
 				if (err != OK) {
 					push_error(vformat(R"(Could not preload resource script "%s".)", p_preload->resolved_path), p_preload->path);
 				}
 			} else {
-				p_preload->resource = ResourceLoader::load(p_preload->resolved_path);
-				if (p_preload->resource.is_null()) {
-					push_error(vformat(R"(Could not preload resource file "%s".)", p_preload->resolved_path), p_preload->path);
+				bool skip_loading = false;
+
+				// Permit cyclic loading of a `PackedScene` if `ResourceLoader` is currently loading it.
+				// If it's not, `skip_loading` will stay false.
+				if (resource_type == "PackedScene") {
```

Comment:
This should not only be `PackedScene`, but also any non-`GDScript` resource, because they too can have scripts that will lead to cyclic loading.